integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py

```diff
@@ -288,3 +190,170 @@ def from_dict(cls, data: Dict[str, Any]) -> "AmazonBedrockChatGenerator":
+            params = {
+                "modelId": self.model,
+                "messages": messages_list,
+                "system": system_prompts,
+                "inferenceConfig": inference_config,
+            }
+            if tool_config:
+                params["toolConfig"] = tool_config
+            if additional_fields:
+                params["additionalModelRequestFields"] = additional_fields
+            callback = streaming_callback or self.streaming_callback
+            if callback:
+                response = self.client.converse_stream(**params)
+                response_stream = response.get("stream")
+                if not response_stream:
+                    msg = "No stream found in the response."
+                    raise AmazonBedrockInferenceError(msg)
+                replies = self.process_streaming_response(response_stream, callback)
+            else:
+                response = self.client.converse(**params)
+                replies = self.extract_replies_from_response(response)
+        except ClientError as exception:
+            msg = f"Could not generate inference for Amazon Bedrock model {self.model} due: {exception}"
+            raise AmazonBedrockInferenceError(msg) from exception
+
+        return {"replies": replies}
+
+    def extract_replies_from_response(self, response_body: Dict[str, Any]) -> List[ChatMessage]:
+        replies = []
+        if "output" in response_body and "message" in response_body["output"]:
+            message = response_body["output"]["message"]
+            if message["role"] == "assistant":
+                content_blocks = message["content"]
+
+                # Common meta information
+                base_meta = {
+                    "model": self.model,
+                    "index": 0,
+                    "finish_reason": response_body.get("stopReason"),
+                    "usage": {
+                        # OpenAI's format for usage for cross ChatGenerator compatibility
+                        "prompt_tokens": response_body.get("usage", {}).get("inputTokens", 0),
+                        "completion_tokens": response_body.get("usage", {}).get("outputTokens", 0),
+                        "total_tokens": response_body.get("usage", {}).get("totalTokens", 0),
+                    },
+                }
+
+                # Process each content block separately
+                for content_block in content_blocks:
+                    if "text" in content_block:
+                        replies.append(ChatMessage.from_assistant(content=content_block["text"], meta=base_meta.copy()))
+                    elif "toolUse" in content_block:
+                        replies.append(
+                            ChatMessage.from_assistant(
+                                content=json.dumps(content_block["toolUse"]), meta=base_meta.copy()
+                            )
+                        )
+        return replies
+
+    def process_streaming_response(
```

Comment:
I would also like to see a unit test for this method, but IDK how tricky it is, so I'll leave it up to you